### ISSUE TITLE
CLI Test Suite

### ISSUE THREAD

**Issue by daemon1024** [publiclab/image-sequencer#1747]
_Follow up meta issue for #1694 and #1718_ 

##### Aim of this suite

Write tests for various components of the CLI aspect of Image Sequencer.

##### Where to add the tests

```
test/cli/*
```

##### How to run these tests

```shell
npm run test-cli
```

#### List of needed tests

- [ ] various kinds of parsing
- [ ] installModule
- [ ] saveSequence
- [ ] sequencerSteps
- [ ] config and steps ( ref #1634 )
- [ ] input/output
- [ ] basic mode

Mention more in the comments if I missed out on something

#### How to implement

https://github.com/publiclab/image-sequencer/blob/main/index.js has an initial implementation.

- create a new `commander` instance
- use `parse` to pass arguments as user.
- use `tape` for other aspects of a test.

@publiclab/is-maintainers @publiclab/is-cli-maintainers @publiclab/is-tests-maintainers

**Comment by jywarren** [comment]
This is great!!

Let's tie it to the documentation for the cli, like, reading the readme
ought to link to corresponding tests demonstrating those features, what do
you think?

We might also reuse test scenarios (input and expected output images) from
other test suites like the basic module tests.

Thanks so much - this is really helpful!!

**Comment by HarshKhandeparkar** [comment]
Should we add something to CONTRIBUTING.md too?

**Comment by jywarren** [comment]
oh, good idea!

**Comment by daemon1024** [comment]
Yeah this is something I felt too. The test aren't documented well in general so documenting the CLI tests along the way will be start :)

**Comment by daemon1024** [comment]
How should I proceed btw? 😅 Open individual issues for each test scenario? Or just directly PRs.

Also should I plan them out in a project or linking them to this issue is enough?

**Comment by HarshKhandeparkar** [comment]
Do whatever is convenient :) But I suppose linking them here is a minimum.

**Comment by HarshKhandeparkar** [comment]
@daemon1024 Is it possible to create a test template so that people don't have to spawn the `commander` instance each time? eg: `/tests/core/templates/`

**Comment by daemon1024** [comment]
That makes sense. Let me try to implement that.

**Comment by daemon1024** [comment]
@HarshKhandeparkar What do you think about #1785. This eliminates the need to spawn commander instance each time. 

P.s. Sorry I got busy with university and couldn't work on it, Now that I have holidays I hope to make some progress on the test-suite.

**Comment by daemon1024** [comment]
We should probably merge https://github.com/publiclab/image-sequencer/pull/1786 before proceeding with the suite.

**Comment by jywarren** [comment]
Great, i left some comments and analysis there although didn't get too far. thanks!

**Comment by daemon1024** [comment]
Since https://github.com/publiclab/image-sequencer/pull/1820 is now merged. I guess we can proceed with is.

**Comment by daemon1024** [comment]
Updated how to implement with the refactored test method.

> https://github.com/publiclab/image-sequencer/blob/main/test/cli/saveSequence.js has an initial implementation.
> 
> - Use `tape` to setup a test.
> - Pass in your required parameters inside `cli` function. https://github.com/publiclab/image-sequencer/blob/0580218c6310486be7ee4275ab85fc24f85bd7eb/test/cli/saveSequence.js#L16

**Comment by daemon1024** [comment]
I will begin working on it and try to introduce a few `fto`'s as well.
